docs/ref/models/relations.txt

```diff
@@ -205,3 +205,88 @@ Related objects reference
        If you use :meth:`~django.db.models.query.QuerySet.prefetch_related`,
        the ``add()``, ``remove()``, ``clear()``, and ``set()`` methods clear
        the prefetched cache.
+
+.. class:: ManyToManyDescriptor
+
+    A "related descriptor" is a descriptor used in a many-to-many
+    related context. This happens for both sides of a
+    :class:`~django.db.models.ManyToManyField` relation::
+
+          class Topping(models.Model):
+              # ...
+              pass
+
+          class Pizza(models.Model):
+              toppings = models.ManyToManyField(Topping)
+
+    In this example, the methods below will be available both on
+    ``Topping.pizza_set`` and on ``Pizza.toppings``.
+
+    .. method:: add_relations(relations, *, assert_no_collisions=False)
```

Comment:
I think it should be noted somewhere here that this won't work with custom through models, unless the fields have a default value set for all non-pk columns?

Reply:
Agreed. Same also holds for methods like `m1.add(*[a, b, c])` I believe. I'll check their docs.